### ISSUE TITLE
Unable to install Gitlab catalog-app from official Helm repository

### ISSUE THREAD

**Issue by renesch-de** [rancher/rancher#16623]
**Rancher versions:**
2.1.1

**Infrastructure Stack versions:**
kubernetes: v1.11.3

**Docker version: (docker version,docker info preferred)**
Server Version: 17.03.2-ce
Storage Driver: aufs
Root Dir: /big/docker/aufs
Backing Filesystem: extfs
Dirs: 196
Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
Volume: local
Network: bridge host macvlan null overlay
Docker Root Dir: /big/docker

**Operating system and kernel: (cat /etc/os-release, uname -r preferred)**
4.13.0-46-generic
Ubuntu 17.10

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
1. Add Catalog for Gitlab
   Catalog URL: https://charts.gitlab.io/
   Branch: master
   Kind: Helm

2. Launch Catalog in desired Project with minimal answers (e.g. for minikube)
   name: gitlab (or whatever the suggestion is)
   Template Version: 1.2.4
   Namespace: gitlab (as suggested by rancher)
   Answers (minikube example):
```
global:
  ingress:
    configureCertmanager: false
    class: "nginx"
  hosts:
    domain: 192.168.99.100.nip.io
    externalIP: 192.168.99.100
certmanager:
  install: false
nginx-ingress:
  enabled: false
gitlab-runner:
  install: false
```

***Results:***

The UI indicates that the Catalog-App is in state "Active" but no rescources except for the namespace are actually created.
When deleting the Catalog-App it stays in state "Removing" indefinetly.

I analyzed the Rancher logs and noticed the following:
```
2018/11/16 09:19:08 [INFO] Creating roleBinding User u-wxv8l Role project-owner
2018/11/16 09:19:08 [INFO] Creating roleBinding User u-wxv8l Role admin
2018/11/16 09:19:08 [INFO] namespaceHandler: addProjectIDLabelToNamespace: adding label field.cattle.io/projectId=p-k2rlq to namespace=gitlab
[main] 2018/11/16 09:19:09 Starting Tiller v2.10.0-rancher5 (tls=false)
[main] 2018/11/16 09:19:09 GRPC listening on :60572
[main] 2018/11/16 09:19:09 Probes listening on :52060
[main] 2018/11/16 09:19:09 Storage driver is ConfigMap
[main] 2018/11/16 09:19:09 Max history per release is 0
[tiller] 2018/11/16 09:19:09 getting history for release gitlab
[storage] 2018/11/16 09:19:09 getting release history for "gitlab"
Release "gitlab" does not exist. Installing it now.
[tiller] 2018/11/16 09:19:09 preparing install for gitlab
[storage] 2018/11/16 09:19:09 getting release history for "gitlab"
[tiller] 2018/11/16 09:19:10 rendering gitlab chart using values
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-role.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-sentinel-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/postgresql/templates/networkpolicy.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-deployment.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/gitlab/charts/mailroom/templates/deployment.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/role.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-daemonset.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/tcp-configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/clusterrole.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/templates/application.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/gitlab/charts/operator/templates/operator_deployment.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-rolebinding.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-slave-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-stats-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/minio/templates/minio_networkpolicy.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-server-statefulset.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/default-backend-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-poddisruptionbudget.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/certmanager-issuer/templates/issuer-job.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-pdb.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/postgresql/templates/secrets.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/registry/templates/deployment.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/gitlab/charts/operator/templates/crd_rbac.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/default-backend-poddisruptionbudget.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-metrics-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-hpa.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/certmanager-issuer/templates/rbac-config.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/rolebinding.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-serviceaccount.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-master-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/serviceaccount.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/registry/templates/ingress.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/registry/templates/pdb.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/registry/templates/hpa.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/certmanager-issuer/templates/cert-manager.yml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/clusterrolebinding.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/registry/templates/service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/udp-configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/registry/templates/configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/headers-configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/controller-service.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/gitlab/charts/operator/templates/gitlab.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/gitlab/charts/operator/templates/crd.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/gitlab/charts/mailroom/templates/configmap.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/redis-ha/templates/redis-sentinel-deployment.yaml" is empty. Skipping.
2018/11/16 09:19:10 info: manifest "gitlab/charts/nginx-ingress/templates/default-backend-deployment.yaml" is empty. Skipping.
[tiller] 2018/11/16 09:19:10 performing install for gitlab
[tiller] 2018/11/16 09:19:10 executing 7 crd-install hooks for gitlab
[tiller] 2018/11/16 09:19:10 hooks complete for crd-install gitlab
[tiller] 2018/11/16 09:19:10 executing 7 pre-install hooks for gitlab
[kube] 2018/11/16 09:19:10 building resources from manifest
```
I find it noteworthy that all templates seem to be empty.
I also looked inside of the Rancher container (exec) and noticed two processes that would not terminate:
```
root 65632 7  0 09:43 ?  00:00:00 helm upgrade --install --namespace gitlab gitlab --set gitlab-runner.install=false,global.hosts.domain=192.168.99.100.nip.io,global.hosts.externalIP=192.168.99.100,global.ingress.class=nginx,global.ingress.configureCertmanager=false,nginx-ingress.enabled=false,certmanager.install=false /tmp/helm-873899453/gitlab
root 65633 7  0 09:43 ?  00:00:00 tiller --listen :60572 --probe :52060
```
I quick look inside of ```/tmp/helm-873899453/gitlab``` suggest that the helm-repo is fully intact with all files present. Here the content for the first templates mentioned:
```
root@ca3fef8107ae:/var/lib/rancher# ls -la /tmp/helm-873899453/gitlab/charts/redis-ha/templates/
total 60
drwxr-xr-x 2 root root 4096 Nov 16 09:19 .
drwxr-xr-x 3 root root 4096 Nov 16 09:19 ..
-rwxr-xr-x 1 root root  434 Nov 16 09:19 NOTES.txt
-rwxr-xr-x 1 root root 1885 Nov 16 09:19 _helpers.tpl
-rwxr-xr-x 1 root root 2268 Nov 16 09:19 redis-configmap.yaml
-rwxr-xr-x 1 root root  904 Nov 16 09:19 redis-master-service.yaml
-rwxr-xr-x 1 root root  451 Nov 16 09:19 redis-pdb.yaml
-rwxr-xr-x 1 root root  374 Nov 16 09:19 redis-role.yaml
-rwxr-xr-x 1 root root  470 Nov 16 09:19 redis-rolebinding.yaml
-rwxr-xr-x 1 root root 2462 Nov 16 09:19 redis-sentinel-deployment.yaml
-rwxr-xr-x 1 root root  421 Nov 16 09:19 redis-sentinel-service.yaml
-rwxr-xr-x 1 root root 5013 Nov 16 09:19 redis-server-statefulset.yaml
-rwxr-xr-x 1 root root  292 Nov 16 09:19 redis-serviceaccount.yaml
-rwxr-xr-x 1 root root  913 Nov 16 09:19 redis-slave-service.yaml
```

The only solution to get rid of the stuck application was to kill the helm process in the Rancher container. The UI cleaned up afterwards and the Catalog-App was not visible anymore. I was also able to delete the gitlab namespace from the UI.

I also tested to install something from the Rancher-Catalog and the Helm-Stable Catalog and both Tests worked fine. When installing the Chart using the same values for Helm outside of Rancher it produces the desired results.

**Comment by deniseschannon** [comment]
@tfiduccia can you try this on v2.2.3? there were some changes in catalog/helm/tiller that might have fixed this.

**Comment by rothomas** [comment]
As it happens I have been trying this on 2.2.3 today with better results, but still no joy.  The namespace is created and the gitlab-ce-postgresql and gitlab-ce-redis containers are deployed and listed as "Active" but with no working pods (the pods are stuck in "Scheduling").  The ingress is defined, but stuck at "Pending".  The app remains in "Deploying" state, and deleting the app now works.

**Comment by davidnuzik** [comment]
Version: v2.2.3, v2.2.4-rc15, latest master (5/28/19) single-node rancher
(tried to reproduce in each)

I'm having a hard time reproducing the issue; perhaps I am doing something wrong.
Although I _did_ reproduce the `info: manifest "gitlab/charts/redis-ha/templates/redis-role.yaml" is empty. Skipping.` (and others) as @rothomas mentioned above; that is concerning to me.

When I did the following I saw workloads/pods get created and the app is provisioned. I did notice that gitlab-minio is missing PV info, etc so I can't get it to become "Active" as I am probably missing important information in my config or my setup. Still, the namespace, pods, and everything gets created for me each time I test.

![image](https://user-images.githubusercontent.com/45179589/58515814-ea552880-815a-11e9-970b-e42c64d3146c.png)

I can also remove the app without any issues. I followed the steps in the issue description very carefully (use the old 1.2.4 template version, use the proper gitlab chart (not ranchers helm chart), try 1.11 k8s, etc).


Some thoughts:
I'm not using the exact same environment as reported. I tried with k8s 1.11.9 (oldest available to me in v2.2.3 rancher) and a few other things and still wasn't able to reproduce the issue except for the issue in logs mentioned earlier.
